### PR TITLE
implicit intents

### PR DESCRIPTION
**Pull request by vinhcotam**
## Related Tickets
- [#Task xxxxx: Content](https://edu-redmine.sun-asterisk.vn/issues/xxxx)
## WHAT
-  Implicit Intents and Fragment
## Evidence (Screenshot or Video)
image
image
image
image
image
image
image
image

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/blob/master/eng/android/coding_convention_android_kotlin.md | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/android/ex_slide5/ImplicitIntents.kt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_implicit_intents)
+        val editTextSearchWeb: EditText = findViewById(R.id.edit_text_search_web);
+        val buttonSearchWeb: Button = findViewById(R.id.button_search_web);
+        val editTextSearchLocation: EditText = findViewById(R.id.edit_text_search_location);
+        val buttonSearchLocation: Button = findViewById(R.id.button_search_location);
+        val editTextShareMessage: EditText = findViewById(R.id.edit_text_share_message);
+        val buttonShareMessage: Button = findViewById(R.id.button_share_message);
+        buttonSearchWeb.setOnClickListener() {
```
Comment (ghost) [inline app/src/main/java/com/sun/android/ex_slide5/ImplicitIntents.kt]: dùng viewBinding

File: app/src/main/res/layout/activity_implicit_intents.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    android:orientation="vertical"
+    android:paddingLeft="10dp"
+    tools:context=".ex_slide5.ImplicitIntents">
+
+    <EditText
+        android:id="@+id/edit_text_search_web"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="20dp"
+        android:minHeight="48dp"
```
Comment (ghost) [inline app/src/main/res/layout/activity_implicit_intents.xml]: dùng file dimens

File: app/src/main/res/layout/activity_implicit_intents.xml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        android:text="@string/defaul_search_location_value"
+        android:id="@+id/edit_text_search_location"
+        />
+
+
+    <Button
+        android:id="@+id/button_search_location"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:backgroundTint="#A9A9A9"
```
Comment (ghost) [inline app/src/main/res/layout/activity_implicit_intents.xml]: dùng file color

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -9,23 +9,37 @@
...
+            <meta-data
+                android:name="android.app.lib_name"
+                android:value="" />
+        </activity>
+        <activity
+            android:name=".ex_slide3.ActivitySendMessage"
+            android:exported="true" />
     </application>
 
 </manifest>
```
Comment (hungvv-1301) [inline app/src/main/AndroidManifest.xml:45]:
```suggestion
</manifest>

```
// endline cuối mỗi file

File: app/src/main/java/com/sun/android/ex_slide5/ImplicitIntents.kt [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+            if (editTextSearchWeb.text.toString().isNullOrEmpty()) {
+                Toast.makeText(this, R.string.text_required_message, Toast.LENGTH_LONG).show()
+            } else {
+                val searchWebIntent: Intent = Uri.parse(editTextSearchWeb.text.toString()).let { webpage ->
+                    Intent(Intent.ACTION_VIEW, webpage)
+                }
+                try {
+                    startActivity(searchWebIntent);
+                    editTextSearchWeb.setText("")
```
Comment (hungvv-1301) [inline app/src/main/java/com/sun/android/ex_slide5/ImplicitIntents.kt:39]:
```suggestion
                    editTextSearchWeb.text.clear()
```

File: app/src/main/java/com/sun/android/ex_slide5/ImplicitIntents.kt [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+            if (editTextSearchLocation.text.toString().isNullOrEmpty()) {
+                Toast.makeText(this, R.string.text_required_message, Toast.LENGTH_LONG).show()
+            } else {
+                val searchMapIntent: Intent =
+                    Uri.parse("google.navigation:q=" + editTextSearchLocation.text.toString()).let { location ->
+                        Intent(Intent.ACTION_VIEW, location)
+                    }
+                try {
+                    startActivity(searchMapIntent);
+                    editTextSearchLocation.setText("")
```
Comment (hungvv-1301) [inline app/src/main/java/com/sun/android/ex_slide5/ImplicitIntents.kt:57]: .

File: app/src/main/java/com/sun/android/ex_slide6/FragmentEx.kt [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    private val binding by lazy {
+        FragmentExBinding.inflate(layoutInflater)
+    }
+    private var buttonOpenClose: Button? = null
+    var isDisplayFragment: Boolean = false
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(binding.root)
+        buttonOpenClose = binding.buttonOpenClose
+        buttonOpenClose!!.setOnClickListener(){
```
Comment (hungvv-1301) [inline app/src/main/java/com/sun/android/ex_slide6/FragmentEx.kt:22]:
```suggestion
        buttonOpenClose?.setOnClickListener(){
```

Comment (hungvv-1301) [reply app/src/main/java/com/sun/android/ex_slide6/FragmentEx.kt:22]: mà sao k dùng luôn binding...
```suggestion
       binding.buttonOpenClose?.setOnClickListener(){
```

Comment (ghost) [inline app/src/main/java/com/sun/android/ex_slide6/FragmentEx.kt:22]: ko dùng assert non null nhé

File: app/src/main/java/com/sun/android/ex_slide6/FragmentEx.kt [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+            }
+        }
+    }
+    private fun displayFragment() {
+        val entireActivityLifecycle: entireActivityLifecycle = entireActivityLifecycle().newInstance()
+        val fragmentManager: FragmentManager = supportFragmentManager
+        val fragmentTransaction: FragmentTransaction = fragmentManager
+            .beginTransaction()
+        fragmentTransaction.add(R.id.fragment_container,entireActivityLifecycle).addToBackStack(null).commit()
+        buttonOpenClose!!.setText(R.string.close);
```
Comment (ghost) [inline app/src/main/java/com/sun/android/ex_slide6/FragmentEx.kt:36]: .

File: app/src/main/java/com/sun/android/ex_slide6/FragmentEx.kt [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        isDisplayFragment = true
+    }
+    private fun closeFragment() {
+        val fragmentManager = supportFragmentManager
+        val entireActivityLifecycle: entireActivityLifecycle? = fragmentManager
+            .findFragmentById(R.id.fragment_container) as entireActivityLifecycle?
+        if (entireActivityLifecycle != null) {
```
Comment (ghost) [inline app/src/main/java/com/sun/android/ex_slide6/FragmentEx.kt:43]: .

File: app/src/main/java/com/sun/android/ex_slide6/FragmentEx.kt (hunk 8 above)
Comment (hungvv-1301) [inline app/src/main/java/com/sun/android/ex_slide6/FragmentEx.kt:36]:
```suggestion
        buttonOpenClose?.setText(R.string.close);
```

File: app/src/main/res/layout/activity_implicit_intents.xml [hunk 10]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
```
Comment (hungvv-1301) [inline app/src/main/res/layout/activity_implicit_intents.xml:7]: format code Ctr+ Alt+ L

File: app/src/main/java/com/sun/android/ex_slide6/entireActivityLifecycle.kt [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val rootView : View =
+            inflater.inflate(com.sun.android.R.layout.fragment_entire_activity_lifecycle, container, false);
+        val radioGroup : RadioGroup = rootView.findViewById(com.sun.android.R.id.radio_group);
+        radioGroup.setOnCheckedChangeListener { group, checkedId ->
+            val radioButton = radioGroup.findViewById<View>(checkedId)
+            val index = radioGroup.indexOfChild(radioButton)
+            val textView = rootView.findViewById<TextView>(com.sun.android.R.id.fragment_header)
```
Comment (ghost) [inline app/src/main/java/com/sun/android/ex_slide6/entireActivityLifecycle.kt:40]: dùng viewBinding đi đừng dùng findViewById nữa

File: app/src/main/res/values/colors.xml [hunk 12]
```diff
@@ -8,4 +8,5 @@
     <color name="black">#FF000000</color>
     <color name="white">#FFFFFFFF</color>
     <color name="grey">#A9A9A9</color>
+    <color name="gold">#FFD700</color>
 </resources>
```
Comment (hungvv-1301) [inline app/src/main/res/values/colors.xml:12]:
```suggestion
</resources>

```

**Pull request merged**